pkg/guacanalytics/patchPlanning.go

```diff
@@ -130,24 +139,36 @@ func SearchDependenciesFromStartNode(ctx context.Context, gqlclient graphql.Clie
 
 }
 
-func caseOnPredicates(ctx context.Context, gqlclient graphql.Client, q *queueValues, neighbor model.NeighborsNeighborsNode, nodeType string) error {
+func caseOnPredicates(ctx context.Context, gqlClient graphql.Client, q *queueValues, neighbor model.NeighborsNeighborsNode, nodeType NodeType) error {
 	// case on predicates and nodeType
 	switch nodeType {
-	case "packageName":
+	case PackageName:
 		switch neighbor := neighbor.(type) {
 		case *model.NeighborsNeighborsIsDependency:
-			err := exploreIsDependencyFromDepPkg(ctx, gqlclient, q, *neighbor)
+			err := exploreIsDependencyFromDepPkg(ctx, gqlClient, q, *neighbor)
 
 			if err != nil {
 				return err
 			}
 		}
-	}
+	case PackageVersion, SourceName:
+		switch neighbor := neighbor.(type) {
+		case *model.NeighborsNeighborsIsOccurrence:
+			exploreIsOccurrenceFromSubject(ctx, gqlClient, q, *neighbor)
+		}
+	case Artifact:
+		switch neighbor := neighbor.(type) {
+		case *model.NeighborsNeighborsHasSLSA:
+			exploreHasSLSAFromArtifact(ctx, gqlClient, q, *neighbor)
```

Comment:
So is the thought process here that I start with a package name or package version and find every dependency, every artifact that they touch? I thought we wanted to go back up the tree to determine the root node that needs to be updated?

Reply:
So the user would input a package version and or package name that is affected by a vulnerability or malware.  Then it would return every dependency or artifact that is also affected by this vulnerability for the purposed of path planning.  Eventually,  I'll want to return some sort of subgraph and blast radius.  I may change exactly the data structure returned though later so it'll be easier to make a visualizer output from.

Reply:
It should be going back up the dependency tree. Not down it.

Reply:
intention is to find the blast radius. So getting subjects from materials and SLSA and getting `pkg` from `depPkg` for IsDependency. Does that make more sense in this context?

Reply:
```
switch nodeType {
	case PackageName:
		switch neighbor := neighbor.(type) {
		case *model.NeighborsNeighborsIsDependency:
			err := exploreIsDependencyFromDepPkg(ctx, gqlClient, q, *neighbor)

			if err != nil {
				return err
			}
		}
	case PackageVersion, SourceName:
		switch neighbor := neighbor.(type) {
		case *model.NeighborsNeighborsIsOccurrence:
			exploreIsOccurrenceFromSubject(ctx, gqlClient, q, *neighbor)
		}
```
Based on the switch statement, it is moving in the right direction. Not an issue.